### ISSUE TITLE
Overwrite file mechanism for conceptual files

### ISSUE THREAD

**Issue by usigrist** [dotnet/docfx#3040]
**DocFX Version Used**: 2.37

**Template used**: `default`

According to the [Specification](https://dotnet.github.io/docfx/spec/docfx_design_spec.html) the overwrite mechanism is not implemented for conceptual files. 

I'm trying to leverage the 'Edit the doc' feature for a non git data source by providing the _docurl_ metadata via a generated overwrite file during my documentation build process. The metadata cannot be part of the conceptual file, since there are different versions of the conceptual file with different edit urls. 

Probably I'm missing an existing docFx feature to make this work. My attempts with overwrite files for conceptual files were not successful.

**Comment by usigrist** [comment]
The `docurl` is different for every conceptual file (identified by `uid`). My understanding of file metadata is that it is applied to all matching files. If this is correct I would need a file metadata entry for every single file which seems quite cumbersome for lots of conceptual files.

**Comment by superyyrrzz** [comment]
Overwrite files is used to add/modify YAML files for API reference documents, as YAML files are generated automatically from source code. Any modification to it will be lost after re-generating YAML files.

For conceptual files, as we already have whole control over it, we have a much easier way to add data to it, like [YAML header](https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html?tabs=tabid-1%2Ctabid-a#yaml-header)

**Comment by usigrist** [comment]
In theory that is true. However I have to maintain multiple versions of the conceptual documentation in parallel. The versions are kept in different branches in different projects in TFS (on prem). Having the `docurl` in the YAML header would mean a rewrite of the YAML header of every conceptual file during the branch process (not really an option). 

Since there is (afaik) no out of the box support for TFS to implement the 'Edit the doc' feature I tried to generate the `docurl` metadata during the document build process using overwrite files. For git repositories this (or something similar) is done by docFx.

The other option is of course to change the YAML header of each conceptual file during the documentation build process. But it would be nice to have a mechanism to add metadata to conceptual files on a per `uid` basis without having to modify every .md file.

**Comment by herohua** [comment]
@usigrist , if your requirement is to set different values for `docurl` based on different branch names, would you try file metadata (in `docfx.json` or in separate JSON files)?
https://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html#323-separated-metadata-files-for-global-metadata-and-file-metadata

It seems the fileMetadata file example there could meet your requirement without writing to Yaml header.

**Comment by usigrist** [comment]
Since the `docurl` is different for every file, file metadata is not really helping. I implemented the Yaml header rewrite now. I'm closing this issue.
